cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -423,12 +424,47 @@ public RunRecordDetail getRun(ProgramRunId id) {
   }
 
   @Override
-  public void addApplication(ApplicationId id, ApplicationSpecification spec) {
+  public void addApplication(ApplicationId id, ApplicationSpecification spec, @Nullable String author,
+                             Long created, @Nullable String changeSummary, @Nullable String parentVersion) {
+    // Generate the creation time for the version of the app.
     TransactionRunners.run(transactionRunner, context -> {
-      getAppMetadataStore(context).writeApplication(id.getNamespace(), id.getApplication(), id.getVersion(), spec);
+      ApplicationMeta latest = getLatest(id.getNamespace(), id.getApplication());
```

Comment:
The call to `getLatest` will run in a different transaction? I think you might want to access the DB via the same transaction.

Reply:
yes modified it